server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.transport;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static org.elasticsearch.common.settings.Setting.boolSetting;
+import static org.elasticsearch.common.settings.Setting.intSetting;
+import static org.elasticsearch.common.settings.Setting.listSetting;
+import static org.elasticsearch.transport.TransportSettings.BIND_HOST_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.PORT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.PUBLISH_HOST_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.PUBLISH_PORT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_ALIVE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_COUNT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_IDLE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_INTERVAL_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_NO_DELAY_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_RECEIVE_BUFFER_SIZE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_REUSE_ADDRESS_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_SEND_BUFFER_SIZE_PROFILE;
+
+/**
+ * Contains the settings and some associated logic for the settings related to the Remote Access port, used by Remote Cluster Security 2.0.
+ */
+public class RemoteClusterPortSettings {
+    private static final Logger logger = LogManager.getLogger(RemoteClusterPortSettings.class);
+
+    public static final String REMOTE_CLUSTER_PROFILE = "_remote_cluster";
+    public static final String REMOTE_CLUSTER_PREFIX = "remote_cluster.";
+    public static final String REMOTE_CLUSTER_SSL_PREFIX = REMOTE_CLUSTER_PREFIX + "ssl.";
+
+    public static final Setting<Boolean> REMOTE_CLUSTER_PORT_ENABLED = boolSetting(
```

Comment:
Yes, but I expect that will be very difficult - Elasticsearch's network layer is initialized at startup and from what I've seen I think it will take significant renovation to make us capable of opening a port without restarting.